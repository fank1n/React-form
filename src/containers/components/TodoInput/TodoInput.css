.todo-name{
    text-align: center;
    color: rgb(0, 0, 0);
}
.todo-input-filed{
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 30px;
    margin-bottom: -20px;
}
.todo-input{
    outline: none;
    width: 30em;
    height: 30px;
    border: 1px solid #ccc;
    border-color: #ccc;
    border-top-left-radius: 3px;
    border-bottom-left-radius: 3px;
    font-size: 20px;
}
.todo-input::placeholder{
    color: rgb(167, 167, 167);
}
.todo-add-btn{
    background-color: #ccc;
    height: 34px;
    width: 100px;
    border: none;
    border-top-right-radius: 3px;
    border-bottom-right-radius: 3px;
    font-size: 20px;
    align-self: flex-start;
}
.todo-add-btn:hover{
    background-color: #2da44e;
    cursor: pointer;
}
