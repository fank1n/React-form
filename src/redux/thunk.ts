// import {getTodos} from "../api/todos";
import INITIALIZE_TODOS from "./actions/todoActions"

export interface ListItem {
    "id": number,
    "description": string,
    "completed": false,
    "favorite": false
}

export const getTodos = () => {
    return async (dispatch:any) => {
        const todos = await fetch ("http://localhost:8000/todo")
        .then((todos) => todos.json)
        .then((todos) => {
            dispatch({
                type: INITIALIZE_TODOS,
                payload: todos,
            })
        }).catch((error) => console.log(error))
    }
}

