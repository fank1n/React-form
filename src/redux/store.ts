import { applyMiddleware, createStore, compose } from 'redux';
// import { configureStore } from "@reduxjs/toolkit";
import todosReducer from "./reducers/addTodoReducer";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
// export const store = configureStore({
//   reducer: todosReducer,
// });

// export const store = createStore(
//   todosReducer, 
//   compose(applyMiddleware(ReduxThunk),     
//   window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
// );
// export const store = createStore(
//   todosReducer,
//   compose(
//       applyMiddleware(thunk),
//   )
// )
export const store = createStore(todosReducer, composeWithDevTools(applyMiddleware(thunk)));

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;