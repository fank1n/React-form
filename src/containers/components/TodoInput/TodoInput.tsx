import { FC } from 'react'
import React, { useState } from "react"
import { useDispatch, useSelector } from "react-redux";
import { AppDispatch, RootState } from "../../../redux/store";
import './TodoInput.css';
import { addTodo } from "../../../redux/reducers/addTodoReducer";
import logo_plus from "../../../icons/plus.svg"

export const TodoInput: FC = () => {
    const [todo, setTodo] = useState<string>('');

    // const todoList = useSelector((state: RootState) => state);
    const dispatch = useDispatch<AppDispatch>();

    const handleChange = (e: any) => {
        setTodo(e.target.value)
    };
    return (
        <>
            <h1 className="todo-name">Список задач</h1>
            <div className="todo-input-filed">
                <div >
                    <input
                        className="todo-input"
                        type="text"
                        name="todoInput"
                        value={todo}
                        placeholder="Запишите что-нибудь"
                        onChange={(e) => handleChange(e)}
                        maxLength = {160}
                    /><br/>
                    {todo.length === 160
                     ?(<span className="input-counter-error">Превышен лимит текста!</span>)
                     :(<span className="input-counter">Доступное количество символов: {160-todo.length}</span>)}
                    
                </div>
                <button className="todo-add-btn" onClick={() => {
                    dispatch(addTodo(todo));
                    setTodo("");
                }}
                ><img className = "input-plus-btn" src={logo_plus} alt="add" width = "30px" height = "30px"/></button>
            </div>
        </>
    )
}
