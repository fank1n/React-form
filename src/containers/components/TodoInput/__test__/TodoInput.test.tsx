import {TodoInput} from '../TodoInput';
import {describe, expect} from '@jest/globals'
import {mount} from "enzyme"
import { Provider } from 'react-redux';
import { store } from '../../../../redux/store';

describe('TodoInput', () => {
    let someBigString = "2qpqoiwuijegpoqweqqpowjf91928rgpqwpgjqpworirgjpqowiejgpoiqwegpoqwhegpoiqwhepgoiqwwqwpeorigjqpwoiegjp9qwehgpquwhg0uq3h4jgqhnpweourghppwewghhqwpoegupqweugpoqwugpw"
    let wrapper:any;
    beforeEach(() => {
      wrapper = mount(<Provider store={store}><TodoInput/></Provider>)
    });
    
    afterEach(() => {
      wrapper.unmount()
    });
  
    it('TodoInput component should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it('should render one input correctly', () => {
        wrapper.find('input')
        expect (wrapper.length).toBe(1)
    });

    it('Input should work correct', () => {
      wrapper.find('input').simulate('change', {target: {value: "Testing"}});
      expect(wrapper.find('[data-test-id="paragraph-for-input-test"]').text()).toBe("Testing")
    });

    it('span should have correct text', () => {
        expect(wrapper.find('span').text()).toBe('Доступное количество символов: 160')
    });

    it('span should be error', () => {
        const someInput = wrapper.find('[className="todo-input"]');
        someInput.simulate("change", {target :{value: someBigString}});
        expect(wrapper.find('[className="input-counter-error"]').text()).toBe("Превышен лимит текста!")
    });
    it('should render one button correctly', () => {
        wrapper.find('button')
        expect (wrapper.length).toBe(1)
    });
  });