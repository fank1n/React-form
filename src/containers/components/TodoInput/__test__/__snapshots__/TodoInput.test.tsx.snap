// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput App component should render correctly 1`] = `ReactWrapper {}`;

exports[`TodoInput TodoInput component should match snapshot 1`] = `ReactWrapper {}`;
